test/integration/servicecidr/migration_test.go

```diff
@@ -0,0 +1,292 @@
+		if svc.Name == "kubernetes" {
+			continue
+		}
+		// wipe the necessary fields so we can recreate the Service
+		svc.ResourceVersion = ""
+		svc.Spec.ClusterIP = ""
+		svc.Spec.ClusterIPs = nil
+		svc.Status = v1.ServiceStatus{}
+		if err := client1.CoreV1().Services(svc.Namespace).Delete(context.Background(), svc.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatalf("got unexpected error: %v", err)
+		}
+		svc, err := client1.CoreV1().Services(svc.Namespace).Create(context.Background(), &svc, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("got unexpected error: %v", err)
+		}
+		if !strings.Contains(svc.Spec.ClusterIP, "10.168.") {
+			t.Fatalf("Service expected to have an ip in range 10.168.0.0/24, got %s", svc.Spec.ClusterIP)
+		}
+	}
+
+	// start second apiserver with the new range and new service cidr controller
+	s2 := kubeapiservertesting.StartTestServerOrDie(t,
+		apiServerOptions,
+		[]string{
+			"--runtime-config=networking.k8s.io/v1alpha1=true",
+			"--service-cluster-ip-range=" + cidr2,
+			"--advertise-address=10.1.1.1",
+			"--disable-admission-plugins=ServiceAccount",
+		},
+		etcdOptions)
+	defer s2.TearDownFn()
+
+	client2, err := clientset.NewForConfig(s2.ClientConfig)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	defer framework.DeleteNamespaceOrDie(client2, ns, t)
+
+	// switch the controller to the new apiserver
+	cancelFn()
+	s1.TearDownFn()
+
+	// ServiceCIDR controller
+	ctx2, cancelFn2 := context.WithCancel(context.Background())
+	defer cancelFn2()
+	informers2 := informers.NewSharedInformerFactory(client2, resyncPeriod)
+	go servicecidrs.NewController(
+		informers2.Networking().V1alpha1().ServiceCIDRs(),
+		informers2.Networking().V1alpha1().IPAddresses(),
+		client2,
+	).Run(ctx2, 5)
+	informers2.Start(ctx2.Done())
+
+	// delete the kubernetes.default service so the old DefaultServiceCIDR can be deleted
+	// and the new apiserver can take over
+	if err := client2.CoreV1().Services(metav1.NamespaceDefault).Delete(context.Background(), "kubernetes", metav1.DeleteOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	// the default serviceCIDR should  be the new one
+	if err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, time.Minute, false, func(ctx context.Context) (bool, error) {
+		cidr, err := client2.NetworkingV1alpha1().ServiceCIDRs().Get(context.TODO(), defaultservicecidr.DefaultServiceCIDRName, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+
+		if cidr.Spec.IPv4 != cidr2 {
+			return false, nil
+		}
+
+		if len(cidr.Finalizers) == 0 {
+			return false, nil
+		}
+
+		for _, condition := range cidr.Status.Conditions {
+			if condition.Type == networkingv1alpha1.ServiceCIDRConditionReady {
+				return condition.Status == metav1.ConditionTrue, nil
+			}
+		}
+		return false, nil
+	}); err != nil {
+		t.Fatalf("waiting for default service cidr ready condition set to true")
```

Comment:
(so on all of these Poll error messages, you're discarding `err` even though the Poll returns an error that isn't "timed out" in some circumstances)

Reply:
yeah, the Fatalf shoudl include the error